src/test/java/net/kaczmarzyk/spring/data/jpa/domain/InTheFutureE2eTest.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2014-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.domain;
+
+import net.kaczmarzyk.E2eTestBase;
+import net.kaczmarzyk.spring.data.jpa.Customer;
+import net.kaczmarzyk.spring.data.jpa.CustomerRepository;
+import net.kaczmarzyk.spring.data.jpa.web.annotation.Spec;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.time.OffsetDateTime;
+
+import static net.kaczmarzyk.spring.data.jpa.CustomerBuilder.customer;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author Konrad Hajduga (Tratif sp. z o.o.)
+ */
+public class InTheFutureE2eTest extends E2eTestBase {
+
+	@Controller
+	private static class InTheFutureSpecController {
+
+		@Autowired
+		CustomerRepository customerRepo;
+
+		@ResponseBody
+		@RequestMapping(value = "/customersWithSpecialOffer")
+		public Object findByCustomersWithAvailableSpecialOffer(
+			@Spec(path="dateOfNextSpecialOffer", spec=InTheFuture.class) Specification<Customer> spec) {
+
+			return customerRepo.findAll(spec);
+		}
+	}
+
+	@Before
+	public void initData() {
+		customer("Barry", "Benson")
+			.nextSpecialOffer(OffsetDateTime.now().plusDays(7))
+			.build(em);
+		customer("Vanessa", "Bloom")
+			.nextSpecialOffer(OffsetDateTime.now().minusDays(7))
+			.build(em);
+	}
+
+	@Test
+	public void findsByNextSpecialOfferFromFuture() throws Exception {
+		mockMvc.perform(get("/customersWithSpecialOffer")
+				.accept(MediaType.APPLICATION_JSON))
+			.andExpect(status().isOk())
+			.andExpect(jsonPath("$").isArray())
+			.andExpect(jsonPath("$[0].firstName").value("Barry"))
+			.andExpect(jsonPath("$[0].lastName").value("Benson"))
+			.andExpect(jsonPath("$[1]").doesNotExist());
+	}
```

Comment:
To maintain consistency it would be better imho to move this test to date-time e2e related classes i.e `LocalDateTimeE2eTest`, `InstantE2eTest` etc.